api/src/utils/get-ip-from-req.ts

```diff
@@ -1,16 +1,32 @@
 import { useEnv } from '@directus/env';
-import type { Request } from 'express';
+import type { IncomingMessage } from 'http';
 import { isIP } from 'net';
+import proxyAddr from 'proxy-addr';
 import { useLogger } from '../logger/index.js';
 
-export function getIPFromReq(req: Request): string | null {
+export function getIPFromReq(req: IncomingMessage): string | null {
 	const env = useEnv();
 	const logger = useLogger();
 
-	let ip = req.ip;
+	let trust: any = env['IP_TRUST_PROXY'];
+
+	// booleans must be passed as a function
+	if (typeof trust === 'boolean') {
+		trust = () => env['IP_TRUST_PROXY'];
+	}
+
+	let ip = proxyAddr(req, trust);
```

Comment:
Could this step be skipped for http requests as it's already handled by https://github.com/directus/directus/blob/298ffaa4b62e8b14a54b37fb707bd6bcad6faeb5/api/src/app.ts#L118 ? 

Does the initial request for ws connection require the proxyAddr logic?

Reply:
As discussed it can be skipped and we now by default use the ip property if present, otherwise fallback to using the `proxyAddr` processing function. This was done via [1918804](https://github.com/directus/directus/pull/25114/commits/191880431b3497e72621b8dc4615e244b07cba03).

The initial http request for the ws connection does require this logic as it is not augmented by express with this value like all of our other requests are.